pkg/clients/handlers.go

```diff
@@ -29,12 +32,27 @@ type PaginationLinksResponse struct {
 	Next  string `json:"next,omitempty"`
 }
 
-func (a *API) RegisterEndpoints(mux *chi.Mux) {
+func (a *API) RegisterEndpoints(mux *chi.Mux) *API {
 	mux.Get("/api/v0/clients", a.ListClients)
 	mux.Post("/api/v0/clients", a.CreateClient)
 	mux.Get("/api/v0/clients/{id}", a.GetClient)
 	mux.Put("/api/v0/clients/{id}", a.UpdateClient)
 	mux.Delete("/api/v0/clients/{id}", a.DeleteClient)
+
+	return a
+}
+
+func (a *API) SetupValidation(v *validation.Validator) *API {
```

Comment:
```suggestion
func (a *API) SetupValidation(v validation.ValidatorInterface) *API {
```

let's use interfaces, happy to use a shared one here instead of a local

Reply:
> let's use interfaces, happy to use a shared one here instead of a local

Just a non-blocking question: why an interface is more appropriate than a struct here? Is it going to have different implementations of the validator registry?